signac/core/synced_collection.py

```diff
@@ -55,8 +55,13 @@ def register(cls, *args):
         """
         if not hasattr(cls, 'registry'):
             cls.registry = defaultdict(list)
+        if not hasattr(cls, 'backend_registry'):
+            cls.backend_registry = []
```

Comment:
Rather than creating these registries here, let's just define the `registry` and the `backend_registry` directly in `SyncedCollection` class. This usage is the exact opposite of what we've been discussing in the validation PR, in which every class needs to know its own validators. Here, there is only one registry, and it is stored at the top of the hierarchy.